File: libparabolic/src/events/downloadaddedeventargs.cpp
#include "events/downloadaddedeventargs.h"

using namespace Nickvision::TubeConverter::Shared::Models;

namespace Nickvision::TubeConverter::Shared::Events
{
    DownloadAddedEventArgs::DownloadAddedEventArgs(int id, const std::filesystem::path& path, const std::string& url, DownloadStatus status)
        : m_id{ id },
        m_path{ path },
        m_url{ url },
        m_status{ status }
    {
        
    }

    int DownloadAddedEventArgs::getId() const
    {
        return m_id;
    }

    const std::filesystem::path& DownloadAddedEventArgs::getPath() const
    {
        return m_path;
    }

    const std::string& DownloadAddedEventArgs::getUrl() const
    {
        return m_url;
    }

    DownloadStatus DownloadAddedEventArgs::getStatus() const
    {
        return m_status;
    }
}

File: Nickvision.Parabolic.Shared/Events/DownloadAddedEventArgs.cs
using Nickvision.Parabolic.Shared.Models;
using System;

namespace Nickvision.Parabolic.Shared.Events;

public class DownloadAddedEventArgs : DownloadEventArgs
{
    public string Path { get; }
    public Uri Url { get; }
    public DownloadStatus Status { get; }

    public DownloadAddedEventArgs(int id, string path, Uri url, DownloadStatus status) : base(id)
    {
        Path = path;
        Url = url;
        Status = status;
    }

    public DownloadAddedEventArgs(Download download) : this(download.Id, download.FilePath, download.Options.Url, download.Status)
    {

    }
}
